### PR TITLE
gnupg: 2.4.5 -> 2.4.7

### LINKED ISSUE #352397: Update request: gnupg 2.4.5 -> 2.4.7
- Package name: gnupg
- Latest released version: 2.4.7

- Current version on the unstable channel: 2.4.5
- Current version on the stable/release channel: 2.4.5

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

-----

Note for maintainers: Please tag this issue in your PR.

---

@fpletz @stigtsp 

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by itepastra**
update gnupg to the newer release

closes #352397 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by afh — CHANGES_REQUESTED** [review]
Thanks for this, @itepastra 💯

Please note that in order to successfully build on `aarch64-darwin` I had to add `NIX_CFLAGS_COMPILE = "-Wno-implicit-function-declaration";`

**Comment by itepastra** [comment]
> Thanks for this, @itepastra 💯
> 
> Please note that in order to successfully build on `aarch64-darwin` I had to add `NIX_CFLAGS_COMPILE = "-Wno-implicit-function-declaration";`

How would I go about fixing this?
Should I just add it to `configureFlags` in the optional `isDarwin` or is there something else

File: pkgs/tools/security/gnupg/24.nix
Comment (afh) [inline pkgs/tools/security/gnupg/24.nix:89]:
```suggestion
  NIX_CFLAGS_COMPILE = lib.optionalString stdenv.hostPlatform.isDarwin "-Wno-implicit-function-declaration";

  configureFlags =
```

File: pkgs/tools/security/gnupg/24.nix
Comment (afh) [inline pkgs/tools/security/gnupg/24.nix:127]: Am I missing something or are the two identical for-loops to "[m]ake libexec tools available in PATH" and one can be deleted?
```suggestion
```

**Review by afh — CHANGES_REQUESTED** [review]
Apologies for not being more specific in my previous review, @itepastra. Please find below a suggestion to fix the darwin build issues and a question regarding `postInstall`

File: pkgs/tools/security/gnupg/24.nix
Comment (itepastra) [reply pkgs/tools/security/gnupg/24.nix:127]: I'm uncertain, but I'll test it without the second loop

**Comment by itepastra** [comment]
no worries @afh, I'm already very happy with the quick reviews.
I've implemented the suggestions, but I don't have a darwin system to attempt a build on.

Now I know what `NIX_CFLAGS_COMPILE` would imply for future times

**Review by afh — CHANGES_REQUESTED** [review]
Thanks for updating this PR, @itepastra. With the latest changes gnupg 2.4.6 builds and runs well on `aarch64-darwin` 👍 

❗ Given that gnupg is such an essential packages and therefore triggers many rebuilds (see labels) please change this PRs merge base to staging (for details see ["Rebasing between branches (i.e. from master to staging)" in CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging)).

Unfortunately properly testing the entirety of this PR using [nixpkgs-review](https://github.com/Mic92/nixpkgs-review) puts too much of a strain on my machine.

**Comment by itepastra** [comment]
I've also tried using nixpkgs-review for this PR, but didn't have time to build 25k+ packages

**Review by afh — APPROVED** [review]
Thanks for targeting `staging` @itepastra. This looks good from my end, curious what the package maintainers have to say.

Yeah staging builds tend to be quite resource intense, please have a bit of patience for other folks to be able to have a close look, test and chime in to ensure this "base"-package continues to work as expected after the update 🙂

**Comment by stigtsp** [comment]
Thanks for the PR! 

Since GnuPG 2.4 minor releases sometimes come with breaking changes, I'd like to review the delta against 2.4.5 a bit and also review the set of patches we have.

**Comment by itepastra** [comment]
@stigtsp have you had time to look at the changes upstream yet?

**Comment by stigtsp** [comment]
@itepastra @MisileLab thanks! I suggest we wait until [freepg](https://gitlab.com/freepg/gnupg) has reviewed the upstream changes in case there are incompat issues

**Comment by itepastra** [comment]
> 2.4.7 released https://git.gnupg.org/cgi-bin/gitweb.cgi?p=gnupg.git;a=blob;f=NEWS;hb=refs/tags/gnupg-2.4.7

should I edit the pr to use 2.4.7 or should I make a new one?

**Comment by stigtsp** [comment]
> should I edit the pr to use 2.4.7 or should I make a new one?

You can update this one :+1:

**Comment by itepastra** [comment]
I've rebased on the current staging branch to get the new CI, no other changes

**Comment by stigtsp** [comment]
@ofborg test gnupg

**Review by stigtsp — COMMENTED** [review]
Doesn't seem to be any significant issue moving to 2.4.7. 

Waiting for local build/test to complete.

**Review by stigtsp — COMMENTED** [review]
LGTM, as long as ofborg-eval passes

**Review by stigtsp — APPROVED** [review]

**Pull request merged**
